build.rs

```diff
@@ -134,52 +217,202 @@ fn fix_config_map(mut config_map: HashMap<String, String>) -> HashMap<String, St
+if "WITH_THREAD" in build_time_vars:
+    print("WITH_THREAD", build_time_vars["WITH_THREAD"])
+if "Py_TRACE_REFS" in build_time_vars:
+    print("Py_TRACE_REFS", build_time_vars["Py_TRACE_REFS"])
+if "COUNT_ALLOCS" in build_time_vars:
+    print("COUNT_ALLOCS", build_time_vars["COUNT_ALLOCS"])
+if "Py_REF_DEBUG" in build_time_vars:
+    print("Py_REF_DEBUG", build_time_vars["Py_REF_DEBUG"])
+print("Py_DEBUG", build_time_vars["Py_DEBUG"])
+print("Py_ENABLE_SHARED", build_time_vars["Py_ENABLE_SHARED"])
+print("LDVERSION", build_time_vars["LDVERSION"])
+print("SIZEOF_VOID_P", build_time_vars["SIZEOF_VOID_P"])
+"#;
+    let output = run_python_script(&find_interpreter()?, &script)?;
+
+    Ok(parse_script_output(&output))
+}
+
+fn starts_with(entry: &DirEntry, pat: &str) -> bool {
+    let name = entry.file_name();
+    name.to_string_lossy().starts_with(pat)
+}
+fn ends_with(entry: &DirEntry, pat: &str) -> bool {
+    let name = entry.file_name();
+    name.to_string_lossy().ends_with(pat)
+}
+
+/// Finds the `_sysconfigdata*.py` file in the library path.
+///
+/// From the python source for `_sysconfigdata*.py` is always going to be located at
+/// `build/lib.{PLATFORM}-{PY_MINOR_VERSION}` when built from source. The [exact line][1] is defined as:
+///
+/// ```py
+/// pybuilddir = 'build/lib.%s-%s' % (get_platform(), sys.version_info[:2])
+/// ```
+///
+/// Where get_platform returns a kebab-case formated string containing the os, the architecture and
+/// possibly the os' kernel version (not the case on linux). However, when installed using a package
+/// manager, the `_sysconfigdata*.py` file is installed in the `${PREFIX}/lib/python3.Y/` directory.
+/// The `_sysconfigdata*.py` is generally in a sub-directory of the location of `libpython3.Y.so`.
+/// So we must find the file in the following possible locations:
+///
+/// ```sh
+/// # distribution from package manager, lib_dir should include lib/
+/// ${INSTALL_PREFIX}/lib/python3.Y/_sysconfigdata*.py
+/// ${INSTALL_PREFIX}/lib/libpython3.Y.so
+/// ${INSTALL_PREFIX}/lib/python3.Y/config-3.Y-${HOST_TRIPLE}/libpython3.Y.so
+///
+/// # Built from source from host
+/// ${CROSS_COMPILED_LOCATION}/build/lib.linux-x86_64-Y/_sysconfigdata*.py
+/// ${CROSS_COMPILED_LOCATION}/libpython3.Y.so
+///
+/// # if cross compiled, kernel release is only present on certain OS targets.
+/// ${CROSS_COMPILED_LOCATION}/build/lib.{OS}(-{OS-KERNEL-RELEASE})?-{ARCH}-Y/_sysconfigdata*.py
+/// ${CROSS_COMPILED_LOCATION}/libpython3.Y.so
+/// ```
+///
+/// [1]: https://github.com/python/cpython/blob/3.5/Lib/sysconfig.py#L389
+fn find_sysconfigdata(cross: &CrossCompileConfig) -> Result<PathBuf> {
+    let mut sysconfig_paths = search_lib_dir(&cross.lib_dir, &cross);
+    if sysconfig_paths.is_empty() {
+        bail!(
+            "Could not find either libpython.so or _sysconfigdata*.py in {}",
+            cross.lib_dir.display()
+        );
+    } else if sysconfig_paths.len() > 1 {
+        bail!(
```

Comment:
Maybe here before bailing, as a last hope try calling `fs::canonicalize` on all the paths and see if they all resolve to the same thing? That'd fix my machine's edge case.

Reply:
fairly easy fix, see 441d7f5